In [ ]:
#--------------------------------------------------
# Imports
#--------------------------------------------------
import numpy as np
import pandas as pd
import sklearn
import matplotlib.pyplot as plt
import yfinance as yf


[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed

        Month  Trading Days
0     January            21
1    February            19
2       March            22
3       April            21
4         May            20
5        June            22
6        July            22
7      August            21
8   September            21
9     October            22
10   November            20
11   December            22
Total trading days in 2020: 253


In [ ]:
#-----------------------------------------------------------------------------
# Calculating the number of trading days in each month of 2020 for AAPL stock
#-----------------------------------------------------------------------------

jan_data = yf.download("AAPL", start="2020-01-01", end="2020-02-01") # downloading the data from the yfinance module for January 2020
jan_trading_days = jan_data.shape[0] # storing the number of rows for January 2020

feb_data = yf.download("AAPL", start="2020-02-01", end="2020-03-01")
feb_trading_days = feb_data.shape[0]

mar_data = yf.download("AAPL", start="2020-03-01", end="2020-04-01")
mar_trading_days = mar_data.shape[0]

apr_data = yf.download("AAPL", start="2020-04-01", end="2020-05-01")
apr_trading_days = apr_data.shape[0]

may_data = yf.download("AAPL", start="2020-05-01", end="2020-06-01")
may_trading_days = may_data.shape[0]

jun_data = yf.download("AAPL", start="2020-06-01", end="2020-07-01")
jun_trading_days = jun_data.shape[0]

jul_data = yf.download("AAPL", start="2020-07-01", end="2020-08-01")
jul_trading_days = jul_data.shape[0]

aug_data = yf.download("AAPL", start="2020-08-01", end="2020-09-01")
aug_trading_days = aug_data.shape[0]    

sep_data = yf.download("AAPL", start="2020-09-01", end="2020-10-01")
sep_trading_days = sep_data.shape[0]

oct_data = yf.download("AAPL", start="2020-10-01", end="2020-11-01")
oct_trading_days = oct_data.shape[0]

nov_data = yf.download("AAPL", start="2020-11-01", end="2020-12-01")
nov_trading_days = nov_data.shape[0]

dec_data = yf.download("AAPL", start="2020-12-01", end="2021-01-01")
dec_trading_days = dec_data.shape[0]

trading_days_2020 = pd.DataFrame({"Month": ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"],
                                   "Trading Days": [jan_trading_days, feb_trading_days, mar_trading_days, apr_trading_days, may_trading_days, jun_trading_days, jul_trading_days, aug_trading_days, sep_trading_days, oct_trading_days, nov_trading_days, dec_trading_days]})
print(trading_days_2020)

total_trading_days = trading_days_2020["Trading Days"].sum()
print(f"Total trading days in 2020: {total_trading_days}")

### Number of Trading Days

The above code calculates a total of 253 trading days in 2020, which is less than the 365 calendar days because the stock markets are only open on weekdays and also close on public hoidays.